### DISCUSSION TITLE
Navigation slow on all GitHub pages using Firefox

### DISCUSSION THREAD

**Discussion by bert-w** [community/community#27334] [Other Feature Feedback, Questions, & Ideas]
Something changed in the past 2 months. Navigation through repositories feels sluggish: going back and forth through browser history is very slow. This seems to be a frontend framework issue. I am experiencing this on Firefox.

**Comment by MPLew-is** [comment]
Experiencing this on Safari some too - typing into code review comments can be extremely sluggish, with characters appearing sometimes with multi-second delays. I don't experience anything similar on other websites.

**Reply by JoffreyLGT** [reply]
I still experience the same issue on Safari more than a year after you reported it.
Is it still your case or did you find a solution?

Disabled all extensions, so can't be related to them.

**Comment by liam-jones-lucout** [comment]
Also affects Vivaldi and Chrome. Github seems to have become very slow across the board recently

**Comment by DanishMughal1222** [comment]
This issue also affects Chrome in addition to Vivaldi. According to recent developments with Github, the platform has become very slow across the board as a result of recent upgrades. Finding this from [SNGPL Connection](https://sngplbill.pk/sngpl-new-connection/)

**Comment by rasikhq** [comment]
I recently had to use Edge and it GitHub seemed to be working better (still abit slow in page loading and not "instant" as it used to be) which is a surprise considering some people are reporting it to be just as slow on chrome which is basically the same technology

**Comment by sug0** [comment]
I can definitely confirm this. Browsing the notification inbox is especially bad.

**Comment by jfullstackdev** [comment]
I'm seeing other issues in Firefox, 

sad to say but Chrome is the one being supported by developers most of the time, the issue is in Firefox if that is not present in Chrome

there is also one issue regarding Codespaces on Firefox, I mentioned that Firefox uses the RAM inefficiently even though Chrome is actually hogging a lot of RAM

**Reply by Efreak** [reply]
> sad to say but Chrome is the one being supported by developers most of the time

This is true, unfortunately.

> the issue is in Firefox if that is not present in Chrome

This, however, is not necessarily true. If the developers aren't testing in Firefox, or have tested but not bothered trying to find out why it doesn't work, or have found out why it doesn't work but not looked up a way around it, then the issue is with the developers. In this case, I absolutely blame GitHub, it's entirely possible to make a website perform well in Firefox. Most severe Firefox deficiencies I'm aware of currently relate to GPU acceleration and indexeddb, neither of which should be relevant here.

**Comment by Clashsoft** [comment]
I can confirm that github is much slower on firefox. I tested with some pages in my profile (Overview, Repositories) and Firefox takes at least twice as long (1s vs 2-5s).

Btw, you added the Pages tag, but that is for "Github Pages" the product. Maybe you should remove that tag so the developers can see this issue in the right category.

**Comment by EdmundsEcho** [comment]
Firefox (my preferred browser for developing and otherwise) is really slow to render the source pages in github. It’s gotten worse in the past few months - freezing with options to stop running the rendering page. Github needs to look into this. Other websites work fine.

**Comment by dakrone** [comment]
I'm also experiencing this recently, running Firefox 121.0b8 on issue pages trying to open the "Labels" dialog to modify the labels for an issue causes the page to freeze for many seconds. Firefox then pops up a warning that a script is slowing down the page. It seems to have changed maybe in the last couple of weeks.

**Comment by Shaverdoff** [comment]
in edge (microsoft ) github open rapidly.... browser war begin?

**Reply by JoffreyLGT** [reply]
Sounds more like everything none-chromium based is not properly tested.

**Comment by Gordon-Dry** [comment]
And still an issue. Just bumpin' for Firefox 128.0.3

**Reply by epiliper** [reply]
Echoing this; the worst offenders seem to be the "repositories" and "profile" pages, at least for my account. 

Even loading this page took ~6 seconds (context: CPU is 7950x3d and RAM is 64GB 5200Mhz).

**Reply by sergiusvet** [reply]
And when I thought my Ryzen 5 5600x might be the issue.

**Comment by mekedron** [comment]
Same issue here

**Comment by robertmark3456789-sudo** [comment]
I've also noticed that navigation feels noticeably slow on all GitHub pages when using Firefox. It's not just isolated to one repo or section—everything from loading issues to page transitions seems sluggish. I even tried clearing the cache and disabling extensions, but no real improvement. Funny enough, other sites work fine, including some heavy ones like[ Castle](https://castleapp.co/), which loads smoothly in the same browser. Hoping GitHub resolves this soon—it’s really affecting workflow.

**Comment by larrasket** [comment]
@github you need to take action on improving the UI performance. It's extremely horrible overall. Loading some pages in a new tab is _much_ faster than waiting for your navigation to complete: https://yoyo-code.com/why-is-github-ui-getting-so-much-slower/ (cc: @chrispat @ashtom)

related discussions https://github.com/orgs/community/discussions/164845,

**Comment by perry-mitchell** [comment]
Still occurring on Firefox 153.0 (64-bit), on Fedora. Multiple machines too. Primarily on review pages. Horrendously unusable performance. Will likely result in us moving away from Github as a repository host.
